src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+                        {
+                            Dictionary<string, FunctionGroup> fptrs;
+                            if (!functionGroups.TryGetValue(descriptor.Assembly, out fptrs))
+                            {
+                                fptrs = new Dictionary<string, FunctionGroup>();
+                                functionGroups[descriptor.Assembly] = fptrs;
+                            }
+
+                            string qualifiedName = descriptor.QualifiedName;
+                            FunctionGroup functionGroup;
+                            if (!fptrs.TryGetValue(qualifiedName, out functionGroup))
+                            {
+                                functionGroup = new FunctionGroup(qualifiedName);
+                                fptrs[qualifiedName] = functionGroup;
+                            }
+                            functionGroup.AddFunctionDescriptor(descriptor);
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        continue;
+                    }
+                }
+
+                if (functionGroups.Count > 0)
+                {
+                    AddZeroTouchNodesToSearch(functionGroups.Values.SelectMany(x => x.Values.Select(g => g)), ref nodeSearchModel);
+                }
+
+                fileInfos.AddRange(FileInfosFromSearchModel(nodeSearchModel));
+                Console.WriteLine($"{fileInfos.Count()} nodes found during scan");
+            }
+
+            return fileInfos;
+        }
+
+        private static List<FunctionDescriptor> GetFunctionDescriptorsFromDll(PathManager pathManager, Assembly asm)
+        {
+            var objectAsmName = typeof(object).Assembly.GetName().Name;
+            var descriptors = new List<FunctionDescriptor>();
+
+            string extension = System.IO.Path.GetExtension(asm.Location).ToLower();
+            if (extension != ".dll" && extension != ".exe")
+            {
+                return new List<FunctionDescriptor>();
+            }
+
+            // Getting ZT imports from CLRModuleTypes
+            DLLModule dllModule = null;
+            dllModule = DLLFFIHandler.GetModuleForInspection(asm);
+
+            dllModule.ScanModule(true);
+            List<CLRModuleType> moduleTypes = CLRModuleType.GetTypes((CLRModuleType mtype) => { return mtype.Module == dllModule; });
+
+            var customizationFile = LibraryCustomizationServices.GetForAssembly(asm.Location, pathManager);
+
+            foreach (var t in moduleTypes)
+            {
+                try
+                {
+                    var className = t.ClassNode.ClassName;
+                    var externalLib = t.ClassNode.ExternLibName;
+
+                    // For some reason mscorelib sometimes gets pass to here, so filtering it away.
+                    if (t.CLRType.Assembly.GetName().Name == objectAsmName ||
```

Comment:
well, there are some types which return object?

Reply:
Right, what im seeing here is that `System.Object` is being passed to here as a ModuleType meaning that all public methods and properties on object will be treated as if they were nodes
![image](https://user-images.githubusercontent.com/13732445/121929692-5eaa9680-cd39-11eb-9a31-5b2e4ea1fdda.png)